stream_utils.py

```diff
@@ -0,0 +1,48 @@
+from amaranth import Module, Signal
+from amaranth.lib.wiring import Component, In, Out
+from amaranth.lib import stream
+
+
+class LimitForwarder(Component):
+    """
+    Forwards a limited number of bytes from one stream to another, then stops.
+
+    TODO: Documentation, parameterization
+    """
+
+    inbound: In(stream.Signature(8))
+    outbound: Out(stream.Signature(8))
+
+    count: In(9)
+    start: In(1)
+    done: Out(1)
+
+    def elaborate(self, _platform):
+        m = Module()
+
+        countdown = Signal(9)
```

Comment:
Worth making the width a parameter?

Reply:
I'd had it as a TODO, but yeah, worthwhile; done in c5b0c47.